src/notebook/notebookKernel.ts

```diff
@@ -250,27 +250,28 @@ export class JuliaKernel {
                     }
                 })
 
+                const outputsPerExecution = new WeakMap<vscode.NotebookCellExecution,{output: vscode.NotebookCellOutput | undefined, name: 'stdout' |'stderr'}>()
                 this._msgConnection.onNotification(
                     notifyTypeStreamoutput,
                     ({ name, data }) => {
-                        if (name === 'stdout') {
-                            const execution = this._currentExecutionRequest
-                            if (execution) {
-                                execution.appendOutput([
-                                    new vscode.NotebookCellOutput([
-                                        vscode.NotebookCellOutputItem.stdout(data),
-                                    ]),
-                                ])
-                            }
-                        } else if (name === 'stderr') {
-                            const execution = this._currentExecutionRequest
-                            if (execution) {
-                                execution.appendOutput([
-                                    new vscode.NotebookCellOutput([
-                                        vscode.NotebookCellOutputItem.stderr(data),
-                                    ]),
-                                ])
-                            }
+                        const execution = this._currentExecutionRequest
+                        if (!execution) {
+                            return
+                        }
+                        if (name === 'stdout' || name === 'stderr') {
+                            // Ensure \r is in a seprate line.
+                            data.split(/(\r)/).forEach((line) => {
```

Comment:
The reason we need to split them is because the assumption with notebooks is 
* \r would come in a separate output from a process
E.g. a progress message (bar) would display some line,
Then the next stream output would contain the `\r`.
Thus the first line of the next stream would be `\r`
Its very unlikely that `\r` would be in the middle of the string, if thats the case, the code that wrote the message might as well update the string and then write the final string to the stream.

I guess we get a combined string here because the julia code is combining them at some point.